src/runtime/strategies/map-slots.ts

```diff
@@ -114,6 +113,7 @@ export class MapSlots extends Strategy {
 
   static specMatch(slotConnection, slot) {
     return slotConnection.slotSpec && // if there's no slotSpec, this is just a slot constraint on a verb
+          slot.spec && // if there is no spec on the slot, it is a hosted slot in the inner arc
```

Comment:
maybe a helper method on the slot, like isInnerSlot? the thing here is that (i think) sometimes `slot` is a recipe-slot and sometimes it is a slot-context... maybe worth making separate methods? anyway, this is outside the scope of this PR, just grumbling)

Reply:
Alice is refactoring in her upcoming patch so that these 2 are handled separately. Let's see what she proposes :)